### DISCUSSION TITLE
How often should we release?

### DISCUSSION THREAD

**Discussion by Rumperuu** [markcheret/footnotes#91] [Ideas]
During the recent release error (see #55), at least two users made the point that the Plugin updates very frequently (see [here](https://wordpress.org/support/topic/version-2-5-9d1-breaks-wp-down/#post-14115987) and [here](https://wordpress.org/support/topic/latest-update-broke-my-site-19/)).

Currently, we push a new release for every bugfix, new feature, etc. with appropriate version incrementing based on the severity of the changes. However, I think it would be worth thinking about how often we want to release going forward.

A couple approaches that I think would make sense:

1. Only release minor version increments (e.g., x.1.x to x.2.x) and above to the SVN. Users could still apply bugfixes provided by us through the Plugin Editor, safe in the knowledge that when the Plugin updates it will included any bugfixes they already have applied to their copy of the Plugin.
1. Keep the ‘Stable Tag’ field on the SVN set to x.x.0 at all times, which would still allow us to push new bugfixes (e.g., x.x.1 to x.x.2) to the SVN for people who want to manually update, whilst also not spamming people with constant fixes for bugs they aren't experiencing.

Thoughts and comments welcomed.

**Comment by lolzim** [comment]
To me, that should be driven by severity of bugfixes first. Important bugfixes SHOULD be released ASAP to me. New features should arrive when mature (and probably not too much per release).
Minor bugfixes COULD fall into your approaches then.

**Comment by markcheret** [comment]
Agreed with @lolzim

**Comment by Rumperuu** [comment]
> To me, that should be driven by severity of bugfixes first. Important bugfixes SHOULD be released ASAP to me. New features should arrive when mature (and probably not too much per release).
> Minor bugfixes COULD fall into your approaches then.

Definitely, important bugfixes (however we define them—number of users affected, severity of impact, etc.) shouldn't be delayed. I was referring (perhaps unclearly) to minor ones that only affect one or two users, or don't cause much disruption, etc. 

That said, it's also possible that we're basing this on faulty assumptions. A couple of users complained that we release too frequently, but they may well not be representative of the majority of users.

Ultimately, though, it's a project decision rather than a technical one, so one for @markcheret — as long as we have something clear and agreed-upon (I'd suggest writing up whatever is decided in the [Wiki](https://github.com/markcheret/footnotes/wiki/)) then we should be good to go.

**Comment by markcheret** [comment]
To move the topic forward:
- Regular feature releases and minor bugfix rollups (if applicable): between bi-weekly and around once a month
- Urgent bugfixes: ASAP
All respecting the release process with automated and manual testing before pushing to SVN (which I want to have automated for releases as soon as feasible)
